pkgs/development/interpreters/python/hooks/fix-python-deps-hook.py

```diff
@@ -0,0 +1,59 @@
+import os
+import sys
+from glob import iglob
+from shutil import copyfile
+from tempfile import NamedTemporaryFile
+
+import requirements
+
+
+requirements_txts = iglob(os.environ.get("requirementsTxt", "requirements*.txt"))
+relax_python_deps = os.environ.get("relaxPythonDeps", "").split()
+remove_python_deps = os.environ.get("removePythonDeps", "").split()
+try:
+    debug = int(os.environ.get("NIX_DEBUG", "0")) >= 7
+except ValueError:
+    debug = False
+
+def debug_msg(*args):
+    if debug:
+        print(f"fixPythonDepsHook:", *args, file=sys.stderr)
+
+
+def fix_deps(requirements_txt):
+    with NamedTemporaryFile(mode="w") as tmp:
+        with open(requirements_txt, mode="r") as f:
+            for line in f.readlines():
```

Comment:
This approach of parsing line by line will not work in one case, that is when someone uses `\` to continue to the next line.

Parsing the whole file should still work, but I decided to parse it line by line first to not lose any information (like comments).

Well, think we will sacrifice the comments for correctness, though.

Reply:
I don't think I have seen that in the wild, yet. Its fine if we just throw an error in this case.

Reply:
I use this in a project of mine actually 😅 . This is why I remember this syntax.

BTW, we can either lose comments and have a more compliant parser or have a non-compliant parser and have comments. I don't think the `requirements.txt` comments will be very useful here in `nixpkgs`, but this is my opinion.

Reply:
I don't really care. What is more convenient.

Reply:
Done.

This change actually made the code more simple.